src/filters/capture.rs

```diff
@@ -219,39 +238,52 @@ mod tests {
     #[test]
     fn write() {
         let config = Config {
-            strategy: Strategy::Suffix,
+            strategy: Strategy::Suffix(Suffix {
+                size: 0,
+                remove: false,
+            }),
             metadata_key: TOKEN_KEY.into(),
-            size: 0,
-            remove: false,
         };
         let filter = capture_bytes(config);
         assert_write_no_change(&filter);
     }
 
     #[test]
     fn end_capture() {
-        let end = Suffix {};
+        let metrics = Metrics::new(&Registry::default()).unwrap();
+        let mut end = Suffix {
+            size: 3,
+            remove: false,
+        };
         let mut contents = b"helloabc".to_vec();
-        let result = end.capture(&mut contents, 3, false);
-        assert_eq!(b"abc".to_vec(), result);
+        let result = end.capture(&mut contents, &metrics).unwrap();
+        assert_eq!(Value::Bytes(b"abc".to_vec().into()), result);
         assert_eq!(b"helloabc".to_vec(), contents);
 
-        let result = end.capture(&mut contents, 3, true);
-        assert_eq!(b"abc".to_vec(), result);
+        end.remove = true;
+
+        let result = end.capture(&mut contents, &metrics).unwrap();
+        assert_eq!(Value::Bytes(b"abc".to_vec().into()), result);
         assert_eq!(b"hello".to_vec(), contents);
     }
 
     #[test]
     fn beginning_capture() {
-        let beg = Prefix {};
+        let metrics = Metrics::new(&Registry::default()).unwrap();
+        let mut beg = Prefix {
+            size: 3,
+            remove: false,
+        };
         let mut contents = b"abchello".to_vec();
 
-        let result = beg.capture(&mut contents, 3, false);
-        assert_eq!(b"abc".to_vec(), result);
+        let result = beg.capture(&mut contents, &metrics);
+        assert_eq!(Some(Value::Bytes(b"abc".to_vec().into())), result);
         assert_eq!(b"abchello".to_vec(), contents);
 
-        let result = beg.capture(&mut contents, 3, true);
-        assert_eq!(b"abc".to_vec(), result);
+        beg.remove = true;
+
+        let result = beg.capture(&mut contents, &metrics);
```

Comment:
I'd like to see a `regex_capture()` test like `end_capture()` and `beginning_capture()`, so we can have a faster interation loop without have to rely on an e2e test to determine if the regex capture is working as expected.